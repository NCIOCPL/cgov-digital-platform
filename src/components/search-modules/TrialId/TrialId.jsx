import React from 'react';
import PropTypes from 'prop-types';
import { Fieldset, TextInput } from '../../atomic';
import './TrialId.scss';

const TrialId = ({ handleUpdate, useValue }) => {
  return (
    <Fieldset
      id="trialid"
      legend="Trial ID"
      helpUrl="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#trialid"
    >
      <TextInput
        action={handleUpdate}
        id="search-trial-id"
        name="trialId"
        value={useValue('trialId')}
        type="text"
        label="Separate multiple IDs with commas or semicolons."
      />
    </Fieldset>
  );
};

TrialId.propTypes = {
  handleUpdate: PropTypes.func,
  useValue: PropTypes.func
};

export default TrialId;
