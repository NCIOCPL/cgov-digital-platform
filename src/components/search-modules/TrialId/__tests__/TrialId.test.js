import React from 'react';
import { render, shallow, mount } from 'enzyme';
import {
  resetDom,
  setupDom
} from '../../../../utilities/testHelpers';
import TrialId from '../TrialId';

const defaultProps = {
}

const setup = (enzymeMethod = render, props = {}) => {
  const component = enzymeMethod(
    <TrialId {...props} />
  );
  return component;
};

describe('TrialId', () => {
  describe('Render', () => {
    let component = null;

    afterEach(() => {
      resetDom();
    });

    it('renders without error', () => {
      component = setup(render, defaultProps);
      expect(component).toMatchSnapshot();
    });
  });

});
