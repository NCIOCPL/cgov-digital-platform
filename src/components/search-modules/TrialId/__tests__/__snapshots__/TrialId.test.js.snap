// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrialId Render renders without error 1`] = `
<fieldset
  class="cts-fieldset "
  id="fieldset--trialid"
>
  <legend
    class="cts-fieldset__legend"
  >
    <span>
      Trial ID
    </span>
  </legend>
  <a
    aria-label="Help"
    class="text-icon-help"
    href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#trialid"
    rel="noopener noreferrer"
    target="_blank"
  >
    ?
  </a>
  <div
    class="cts-fieldset__body"
  >
    <div
      class=""
    >
      <label
        class="cts-label"
        for="search-trial-id"
        id="search-trial-id-label"
      >
        Separate multiple IDs with commas or semicolons.
      </label>
      <input
        aria-labelledby="search-trial-id-label"
        aria-required="false"
        class="cts-input null"
        id="search-trial-id"
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
  </div>
</fieldset>
`;
